pkg/controller/pod_driver.go

```diff
@@ -315,21 +374,25 @@ func (p *PodDriver) podReadyHandler(ctx context.Context, pod *corev1.Pod) (recon
 		}
 	}
 
-	// recovery for each target
-	for k, target := range pod.Annotations {
-		if k == util.GetReferenceKey(target) {
-			mi := p.mit.resolveTarget(target)
-			if mi == nil {
-				klog.Errorf("pod %s target %s resolve fail", pod.Name, target)
-				continue
-			}
+	_ = doWithinTime(ctx, func() error {
```

Comment:
This function can not be executed in concurrence!
Use doWithinTime will case some unexpeced things happen.